docs/contracts.rst

```diff
@@ -1023,23 +1048,33 @@ When the inheritance results in a contract with a function and a modifier of the
 This error is produced also by an event and a modifier of the same name, and a function and an event of the same name.
 As an exception, a state variable getter can override a public function.
 
+.. index:: ! abstractions
+
+.. _abstractions:
+
+************
+Abstractions
```

Comment:
Does this section introduce this term? If yes, is it intended to capture both abstract base contracts and interfaces? I'm not sure we should do this as it might be too easily confused with "abstract contract". I would say an interface is a special kind of abstract base contract.

Reply:
Yes @chriseth see the PR description for motivation.

Reply:
Also I'm not sure I understand the difference you're drawing between an "abstract contract" and an "abstract base contract", can you explain? An abstract contract cannot be instantiated and therefore makes no sense unless it's a base contract too.

Reply:
Not sure we need this section in its current form.

How about having a title like "Special contracts" / "Special contract kinds" / "Special contract types" (or something better you can come up with)?

It would only need a sentence stating that some contracts cannot be compiled, but can be used as bases for other contracts.

Reply:
Maybe the common section was not a good idea after all so I've dropped it